packages/webapp/src/components/DateRangeSelector/index.jsx

```diff
@@ -0,0 +1,198 @@
+    { value: rangeOptions.LAST_WEEK, label: t('DATE_RANGE_SELECTOR.LAST_WEEK') },
+    { value: rangeOptions.THIS_MONTH, label: t('DATE_RANGE_SELECTOR.THIS_MONTH') },
+    { value: rangeOptions.LAST_MONTH, label: t('DATE_RANGE_SELECTOR.LAST_MONTH') },
+    { value: rangeOptions.CUSTOM, label: t('DATE_RANGE_SELECTOR.CUSTOM_RANGE') },
+  ];
+
+  // set defaultCustomDateRange if exists
+  useEffect(() => {
+    if (defaultDateRangeOptionValue !== rangeOptions.CUSTOM) {
+      return;
+    }
+    const { [FROM_DATE]: defaultFromDate, [TO_DATE]: defaultToDate } = defaultCustomDateRange;
+
+    // if the range does not have both FROM_DATE and TO_DATE, reset the option to "Year to date"
+    if (!defaultFromDate || !defaultToDate) {
+      setSelectedDateRangeOption(options[0]);
+      return;
+    }
+
+    setCustomFromDate(defaultFromDate);
+    setCustomToDate(defaultToDate);
+    setValidRange(defaultFromDate <= defaultToDate);
+  }, []);
+
+  useEffect(() => {
+    setAreValidDates(customFromDate?.isValid() && customToDate?.isValid());
+  }, [customFromDate, customToDate]);
+
+  const formatOptionLabel = (data, formatOptionLabelMeta) => {
+    if (formatOptionLabelMeta.context === 'menu') {
+      const selected = formatOptionLabelMeta.selectValue[0]?.value === data.value;
+      return <span className={selected ? styles.selectedMenuOptionInMenu : ''}>{data.label}</span>;
+    }
+
+    let formattedOption = data.label;
+    let className = '';
+
+    if (data.value === rangeOptions.CUSTOM) {
+      if (isValid) {
+        formattedOption = [customFromDate, customToDate]
+          .map((date) => date.format('YYYY-MM-DD'))
+          .join(' - ');
+      } else {
+        formattedOption = 'yyyy-mm-dd - yyyy-mm-dd';
+        className = styles.invalid;
+      }
+    }
+
+    return (
+      <div className={styles.window}>
+        <Calendar />
+        <span className={clsx(styles.windowValue, className)}>{formattedOption}</span>
+      </div>
+    );
+  };
+
+  const setSelectedDateRangeOption = (option) => {
+    selectRef.current.setValue(option);
+  };
+
+  const clearCustomDateRange = () => {
+    setCustomFromDate(undefined);
+    setCustomToDate(undefined);
+  };
+
+  const onClickAway = () => {
+    if (dateRange.option !== dateRangeOptions.CUSTOM) {
+      return;
+    }
+    if (!isValid) {
+      setSelectedDateRangeOption(options[0]);
+    }
+    setIsCustomDatePickerOpen(false);
+  };
+
+  const onBack = () => {
+    setIsCustomDatePickerOpen(false);
+    selectRef.current.focus();
+  };
+
+  const changeStartDate = (date) => {
+    setValidRange(date <= customToDate);
+    setCustomFromDate(date);
+    changeDateMethod('start', date);
+  };
+
+  const changeEndDate = (date) => {
+    setValidRange(customFromDate <= date);
+    setCustomToDate(date);
+    changeDateMethod('end', date);
+  };
+
+  return (
+    <ClickAwayListener onClickAway={onClickAway}>
+      <div className={styles.wrapper}>
+        <ReactSelect
```

Comment:
Feel free to ignore this :) 

I think the only question I have is do we have a different non-form element other than ReactSelect?

This is related to my question of tech debt for Form elements vs View elements. I can see value in separating the use of form elements that can be used react-hook-form. And view elements like filters since this (mostly) acts like a filter for data. (But I do get that an argument can be made that it is also a form element for document export -- that does not use react-hook-form)

Reply:
There is an input that is a non-form element that is used for the custom date range. (`FromToDateContainer` -> `DateContainer`)
<img width="495" alt="Screenshot 2023-10-12 at 11 12 25 PM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/2f43e0e5-d10c-4199-b266-552e5e104ab1">
Now I feel that you are asking if we have a select element for non-form... - I don't think we do.

I thought I understood your question at that time, but now I am confused... you don't want us to use the same ReactSelect component when it's not part of a form?

Reply:
Yes you are right that is what I was asking! Since we don't have one now I don't think anything should be done. All I could find is DropDown and it is unused.

I am wondering if that makes sense that we _should_ then maybe have a non-form react select in the future. One for taking/managing user input data into a RHF, one for filtering data to a view from redux. And should elements like this DateRangeSelector live under /components/Filter/DateRangeSelector or something?

Reply:
Thank you for clarifying! By taking a quick look, I don't feel like a ReactSelect component for non-form will be very different from the existing ReactSelect (I need to look into the taking/managing user input data part), but I think this could be an interesting discussion!

I initially created this component using `Form/DateRangePicker` that takes properties from RHF without knowing the difference from `DateRangeSelector, and I ended up almost re-doing to switch to the one without RHF 😢  So in short, I agree with setting a clear rule (like where to place files) for forms and non-forms and documenting it!